SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -271,13 +269,15 @@ public async Task<IActionResult> AskForPostcode_Post(AskForPostcodeViewModel vie
             {
                 return await AskForPostcode_Get(viewModel.Reference);
             }
-            
-            var forwardArgs = questionFlowService.ForwardLinkArguments(QuestionFlowPage.AskForPostcode, propertyData);
+
+            var nextStep = questionFlowService.NextStep(QuestionFlowStep.AskForPostcode, propertyData);
+            var forwardArgs = GetActionArgumentsForQuestion(nextStep, viewModel.Reference);
 
             return RedirectToAction(
                 forwardArgs.Action,
                 forwardArgs.Controller,
-                // TODO: seabeta-579 we'd rather use the values from the question flow service
+                // It would be nice to find a way to get GetActionArgumentsForQuestion() to include the postcode and
+                // house number.
                 new { reference = viewModel.Reference, postcode = viewModel.Postcode, number = viewModel.HouseNameOrNumber });
```

Comment:
First thing I could think of is passing both values as parameters to `GetActionArgumentsForQuestion()`. The logic might get complicated because anonymous object creation is quite strict regarding types, see https://stackoverflow.com/questions/27702704/include-properties-based-on-conditions-in-anonymous-types...